src/components/Account/AccountModule.js

```diff
@@ -1,145 +1,81 @@
+import { Button, GU, IconConnect, useViewport } from '@aragon/ui'
+import { shortenAddress } from '../../lib/web3-utils'
 import AccountButton from './AccountButton'
-import HeaderPopover from '../Header/HeaderPopover'
+import AccountPopover from './AccountPopover'
 import ScreenConnected from './ScreenConnected'
 import ScreenConnecting from './ScreenConnecting'
 import ScreenError from './ScreenError'
 import ScreenProviders from './ScreenProviders'
 
 const SCREENS = [
-  {
-    id: 'providers',
-    title: 'Use account from',
-    height:
-      4 * GU + // header
-      (12 + 1.5) * GU * Math.ceil(getUseWalletProviders().length / 2) + // buttons
-      7 * GU, // footer
-  },
-  {
-    id: 'connecting',
-    title: 'Use account from',
-    height: 38 * GU,
-  },
-  {
-    id: 'connected',
-    title: 'Active account',
-    height: 22 * GU,
-  },
-  {
-    id: 'error',
-    title: 'Connection error',
-    height: 50 * GU,
-  },
+  { id: 'providers', title: 'Use account from' },
+  { id: 'connecting', title: 'Use account from' },
+  { id: 'connected', title: 'Active account' },
+  { id: 'error', title: 'Connection error' },
 ]
 
 function AccountModule() {
-  const buttonRef = useRef()
-  const wallet = useWallet()
   const [opened, setOpened] = useState(false)
-  const [animate, setAnimate] = useState(false)
   const [activatingDelayed, setActivatingDelayed] = useState(false)
-  const [activationError, setActivationError] = useState(null)
-  const popoverFocusElement = useRef()
-
+  const buttonRef = useRef()
   const { below } = useViewport()
   const compactMode = below('medium')
+  const wallet = useWallet()
+  const { account, connector, error, status } = wallet
 
-  const { account, activating } = wallet
-
-  const clearError = useCallback(() => setActivationError(null), [])
-
+  const open = useCallback(() => setOpened(true), [])
   const toggle = useCallback(() => setOpened(opened => !opened), [])
 
-  const handleCancelConnection = useCallback(() => {
-    wallet.deactivate()
-  }, [wallet])
-
-  const activate = useCallback(
-    async providerId => {
-      try {
-        await wallet.activate(providerId)
-      } catch (error) {
-        setActivationError(error)
-      }
-    },
-    [wallet]
-  )
-
-  // Don’t animate the slider until the popover has opened
   useEffect(() => {
-    if (!opened) {
-      return
-    }
-    setAnimate(false)
-    const timer = setTimeout(() => {
-      setAnimate(true)
-    }, 0)
-    return () => clearTimeout(timer)
-  }, [opened])
+    let timer
 
-  // Always show the “connecting…” screen, even if there are no delay
-  useEffect(() => {
-    if (activationError) {
+    if (status === 'error') {
       setActivatingDelayed(null)
     }
 
-    if (activating) {
-      setActivatingDelayed(activating)
-      return
+    if (status === 'connecting') {
+      setActivatingDelayed(connector)
+      timer = setTimeout(() => {
+        setActivatingDelayed(null)
+      }, 400)
```

Comment:
This is something noteworthy: with the removal of the [`activating` escape hatch on error](https://github.com/aragon/court-dashboard/blob/master/src/components/Account/AccountModule.js#L187-L191), the previous implementation of the simulated delays was causing duplicated animations on connection and on disconnection. To avoid this, I had to move the timeout only on the case where we want to actually set it to null for the screen to change, which is the `connecting` case.